lib/pbench/server/database/models/datasets.py

```diff
@@ -632,6 +632,17 @@ def update(self):
             Database.db_session.rollback()
             raise DatasetSqlError("updating", self.controller, self.name) from e
 
+    @staticmethod
+    def query(md5: str) -> "Dataset":
+        """
+        Query dataset object based on md5 of the run document
+        """
+        try:
+            dataset = Database.db_session.query(Dataset).filter_by(md5=md5).first()
```

Comment:
It would be "fun" to generate an `_or` filter across a set of columns specified via `kwargs`. For some definition of "fun".

Unfortunately the default if specifying multiple columns is `_and`, which we wouldn't usually want. I think. Then again, the most likely use for multiple filters would be exclusive rather than combinatorial, and the `kwargs` just allow us to specify any one of the set, in which case `_and` vs `_or` doesn't matter ... 🙄